routes/suggestActivity.js

```diff
@@ -0,0 +1,51 @@
+const express = require("express");
+
+// Import file system
+const { writeFile } = require("node:fs/promises");
+
+// Import mongoose model
+const suggestionModel = require("../models/suggestion");
+
+// Import data from suggestions.json so it can be added to arrayOfSuggestedIdeas
+const arrayOfSuggestedIdeas = require("../suggestions.json");
+
+const { isActivityValid } = require("../util");
+
+const app = express();
+
+// Allow users to suggest new activities via http POST requests on /suggest
+app.post("/suggest", async function (request, response) {
+  // The suggested activity is stored in the body of the http request
+  // const { body } = request;
+  // const suggestedActivity = body;
+
+  // The suggested activity is stored in the mongodb database
+  const suggestedActivity = new suggestionModel(request.body);
+
+  // Calling the function to check that the object passed to the funtion is valid
+  const isValid = isActivityValid(suggestedActivity);
```

Comment:
The creation of a MongoDB model object should be done only after we know that the request body is valid
```suggestion
  // Calling the function to check that the object passed to the function is valid
  const isValid = isActivityValid(request.body);
  
    // create an instance of a MongoDB model to be stored to db
  const suggestedActivity = new suggestionModel(request.body);
```